bpf/lib/nat.h

```diff
@@ -602,18 +602,15 @@ static __always_inline int snat_v4_rewrite_egress(struct __ctx_buff *ctx,
 				return DROP_CSUM_L4;
 #endif  /* ENABLE_SCTP */
 			case IPPROTO_ICMP: {
-				__be32 from, to;
-
 				if (ctx_store_bytes(ctx, off +
 						    offsetof(struct icmphdr, un.echo.id),
 						    &state->to_sport,
 						    sizeof(state->to_sport), 0) < 0)
 					return DROP_WRITE_ERROR;
-				from = tuple->sport;
-				to = state->to_sport;
-				flags = 0; /* ICMPv4 has no pseudo-header */
-				sum_l4 = csum_diff(&from, 4, &to, 4, 0);
-				csum.offset = offsetof(struct icmphdr, checksum);
+				if (l4_csum_replace(ctx, off + offsetof(struct icmphdr, checksum),
+						    tuple->sport,
+						    state->to_sport, 0) < 0)
```

Comment:
Do we need to specify the number of bytes (2) on the lowest four bits of _flags_(the last parameter in bpf_l4_csum_replace)?

Reply:
Yes Interesting point I'm wondering whether we do right in other places as-well. Let me see...

Reply:
Thank you for this review I fixed it.